### DISCUSSION TITLE
How do I upload a guild avatar? (solved)

### DISCUSSION THREAD

**Discussion by dragos240** [Rapptz/discord.py#9580] [Q&A] [answered]
`Member` has an `edit` method but if I try to specify an avatar with `avatar=bytesObject`, it gives me `Member.edit() got an unexpected keyword argument 'avatar'`. I know I can use `ClientUser.edit` to update the avatar of the bot `User`, but I don't see a way to upload a guild avatar. Is there a way or is this something bots do not support?

**Comment by No767** [comment]
`ClientUser` refers to your bot. `Member` refers to a member of a guild. Ofc you can't change someone's pfp using a bot, since that would be extremely dangerous. Plus, [`Member.edit`](https://discordpy.readthedocs.io/en/stable/api.html#discord.Member.edit) does not have a kwarg called `avatar`.

**Reply by dragos240** [reply]
I realize that. I'm talking about my own bot user. Normal users have the ability to upload an avatar for a specific guild for their user. I'm asking if the same functionality is available for bots in Discord.py

**Reply by No767** [reply]
Short answer for that would be no. You can't give a bot a guild avatar. The avatar is set globally.

**Reply by dragos240** [reply]
Is there a piece of documentation that specifies this? Usually if there's a discrepancy between abilities between bots and normal users, it's documented somewhere.

**Reply by No767** [reply]
I don't recall said documentation. It's explicitly clear that the capabilities of uploading a guild avatar for bots don't exist. Ddocs (discord docs) made no mention of this capability for bots

**Reply by dragos240** [reply, marked as answer]
You responded that it was a Discord Intro feature on the Discord.py Discord, which explains it. Closing.
